src/main/java/org/tailormap/api/drawing/DrawingService.java

```diff
@@ -431,12 +434,29 @@ private void canCreateDrawing(@NonNull Authentication authentication) throws Res
    */
   private void canReadDrawing(@NonNull Drawing drawing, @NonNull Authentication authentication)
       throws ResponseStatusException {
-    // TODO validate this method
     boolean isAuthenticated = !(authentication instanceof AnonymousAuthenticationToken);
-
     boolean canRead =
         switch (drawing.getAccess()) {
-          case PRIVATE -> isAuthenticated && Objects.equals(authentication.getName(), drawing.getCreatedBy());
+          case PRIVATE -> {
+            if (isAuthenticated) {
+              if (Objects.equals(authentication.getName(), drawing.getCreatedBy())) {
+                // is drawing owner
+                yield true;
+              }
+              if (authentication.getPrincipal() instanceof TailormapUserDetails userDetails) {
+                // check if the user has either the `drawings-admin` or `drawings-read-all` property set
+                for (AdminAdditionalProperty ap : userDetails.getAdditionalProperties()) {
+                  if (ap.getKey().equals("drawings-admin")
+                      || ap.getKey().equals("drawings-read-all")) {
```

Comment:
perhaps we should create a constants class for key that are used AdminAdditionalProperty in such as `drawings-admin` and `drawings-read-all` and some others maybe in use in eg https://github.com/B3Partners/planmonitor-wonen-api/blob/main/src/main/java/nl/b3p/planmonitorwonen/api/model/auth/PlanmonitorAuthentication.java#L38 to prevent conflicts

Reply:
perhaps adding them to the `AdminAdditionalPropertyHelper`